components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/processor/SAMLLogoutRequestProcessor.java

```diff
@@ -0,0 +1,243 @@
+                        SUCCESS_CODE, null);
+                samlMessageContext.setResponse(SSOUtils.encode(SSOUtils.marshall(logoutResp)));
+                samlMessageContext.setAcsUrl(logoutResp.getDestination());
+            }
+            return buildResponseForFrameworkLogout(samlMessageContext);
+
+        } catch (SAMLSSOException e) {
+            throw new SAMLIdentityException("Error when processing the Logout Request.", e);
+        }
+    }
+
+    /**
+     * Build the authentication request for the framework logout.
+     *
+     * @param samlMessageContext {@link SAMLMessageContext} object which holds details on logout flow.
+     * @return FrameworkLogoutResponse.FrameworkLogoutResponseBuilder instance.
+     */
+    private FrameworkLogoutResponse.FrameworkLogoutResponseBuilder buildResponseForFrameworkLogout
+            (SAMLMessageContext<String, String> samlMessageContext) {
+
+        IdentityRequest identityRequest = samlMessageContext.getRequest();
+        Map<String, String[]> parameterMap = identityRequest.getParameterMap();
+
+        AuthenticationRequest authenticationRequest = new AuthenticationRequest();
+        authenticationRequest.appendRequestQueryParams(parameterMap);
+        if (identityRequest.getHeaderMap() != null) {
+            identityRequest.getHeaderMap().forEach(authenticationRequest::addHeader);
+        }
+        authenticationRequest.setTenantDomain(samlMessageContext.getTenantDomain());
+        authenticationRequest.setRelyingParty(getRelyingPartyId(samlMessageContext));
+        authenticationRequest.setType(getType(samlMessageContext));
+        try {
+            authenticationRequest.setCommonAuthCallerPath(URLEncoder.encode(getCallbackPath(samlMessageContext),
+                    StandardCharsets.UTF_8.name()));
+        } catch (UnsupportedEncodingException e) {
+            throw FrameworkRuntimeException.error("Error occurred while URL encoding callback path: " +
+                    getCallbackPath(samlMessageContext), e);
+        }
+        authenticationRequest.addRequestQueryParam(LOGOUT, new String[]{TRUE});
+        authenticationRequest.addRequestQueryParam(SESSION_ID, new String[]{samlMessageContext.getSessionID()});
+
+        AuthenticationRequestCacheEntry authRequest = new AuthenticationRequestCacheEntry(authenticationRequest);
+        String sessionDataKey = UUIDGenerator.generateUUID();
+        authRequest.setValidityPeriod(TimeUnit.MINUTES.toNanos(IdentityUtil.getOperationCleanUpTimeout()));
+        FrameworkUtils.addAuthenticationRequestToCache(sessionDataKey, authRequest);
+
+        InboundUtil.addContextToCache(sessionDataKey, samlMessageContext);
+
+        FrameworkLogoutResponse.FrameworkLogoutResponseBuilder responseBuilder =
+                new FrameworkLogoutResponse.FrameworkLogoutResponseBuilder(samlMessageContext);
+        responseBuilder.setContextKey(sessionDataKey);
+        responseBuilder.setCallbackPath(getCallbackPath(samlMessageContext));
```

Comment:
since `getCallbackPath(samlMessageContext)` is called in multiple places in this method, better to have it assigned to a variable and reuse

Reply:
Fixed with PR https://github.com/wso2-extensions/identity-outbound-auth-samlsso/pull/133